backend/src/main/java/com/redhat/cloud/notifications/db/EndpointResources.java

```diff
@@ -70,24 +90,29 @@
         });
     }
 
-    public Uni<List<Endpoint>> getEndpointsPerType(String accountId, Set<EndpointType> type, Boolean activeOnly, Query limiter, boolean useStatelessSession) {
-        return commonStateSessionFactory.withSession(useStatelessSession, session -> {
-            // TODO Modify the parameter to take a vararg of Functions that modify the query
-            // TODO Modify to take account selective joins (JOIN (..) UNION (..)) based on the type, same for getEndpoints
-            Query.Limit limit = limiter == null ? null : limiter.getLimit();
-            Query.Sort sort = limiter == null ? null : limiter.getSort();
+    public Uni<List<Endpoint>> getEndpointsPerCompositeType(String accountId, Set<CompositeEndpointType> type, Boolean activeOnly, Query limiter, boolean useStatelessSession) {
+        Query.Limit limit = limiter == null ? null : limiter.getLimit();
+        Query.Sort sort = limiter == null ? null : limiter.getSort();
+        return commonStateSessionFactory.withSession(useStatelessSession, session -> queryBuilderEndpointsPerType(accountId, type, activeOnly)
+                .limit(limit)
+                .sort(sort)
+                .build(session::createQuery)
+                .getResultList()
+                .onItem().call(endpoints -> loadProperties(endpoints, useStatelessSession)));
+    }
 
-            return queryBuilderEndpointsPerType(accountId, type, activeOnly)
-                    .limit(limit)
-                    .sort(sort)
-                    .build(session::createQuery)
-                    .getResultList()
-                    .onItem().call(this::loadProperties);
-        });
+    public Uni<List<Endpoint>> getEndpointsPerType(String accountId, Set<EndpointType> type, Boolean activeOnly, Query limiter, boolean useStatelessSession) {
```

Comment:
```suggestion
    private Uni<List<Endpoint>> getEndpointsPerType(String accountId, Set<EndpointType> type, Boolean activeOnly, Query limiter, boolean useStatelessSession) {
```

Reply:
I deleted this method as part of a cleanup